### PR TITLE
fix: typings

### PR DESCRIPTION
**Pull request by Tolfx**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -36,11 +36,11 @@ export class AsyncQueue {
 	 * ```
 	 * @returns {Promise}
 	 */
-	public new(): Promise<void> {
+	public new(): AsyncPromise["promise"] {
 		const length = this.promises.length;
 		const next = length ? this.promises[length - 1].promise : Promise.resolve();
 
-		let resolve: () => void;
+		let resolve: AsyncPromise["resolve"];
```
Comment (AngeloCore) [inline src/index.ts]: this is undefined sometimes

File: src/index.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export class AsyncQueue {
 	 * Resolving the next promise
 	 * @returns {boolean} If there is a next promise, it will return `true`, if there is no, `false`
 	 */
-	public next(): boolean {
+	public next(): Boolean {
```
Comment (AngeloCore) [inline src/index.ts]: why not `boolean`?

**Review by AngeloCore — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (Tolfx) [reply src/index.ts]: More of a preference of mine, Boolean is an object and boolean is a primitive type. Usually gotten issues with `boolean` from compilers, but probably fixed, so just something I do automatic when I see it.

Comment (AngeloCore) [reply src/index.ts]: ok

File: src/index.ts (hunk 1 above)
Comment (Tolfx) [reply src/index.ts]: Alrighty.

**Review by AngeloCore — APPROVED** [review]

**Pull request merged**